label_maker/package.py

```diff
@@ -86,16 +96,36 @@ def package_directory(dest_folder, classes, imagery, ml_type, seed=False, train_
         elif ml_type == 'segmentation':
             y_vals.append(labels[tile][..., np.newaxis])  # Add grayscale channel
 
-    # split into train and test
-    split_index = int(len(x_vals) * train_size)
-
     # convert lists to numpy arrays
     x_vals = np.array(x_vals, dtype=np.uint8)
     y_vals = np.array(y_vals, dtype=np.uint8)
 
+    x_vals_split_lst = np.split(x_vals,
+                                [int(split_vals[0] * len(x_vals)), int((split_vals[0] + split_vals[1]) * len(x_vals))])
+
+    if len(x_vals_split_lst[-1]) == 0:
+        x_vals_split_lst = x_vals_split_lst[:-1]
+
+    y_vals_split_lst = np.split(y_vals,
+                                [int(split_vals[0] * len(y_vals)), int((split_vals[0] + split_vals[1]) * len(y_vals))])
+
+    if len(y_vals_split_lst[-1]) == 0:
+        y_vals_split_lst = y_vals_split_lst[:-1]
+
     print('Saving packaged file to {}'.format(op.join(dest_folder, 'data.npz')))
-    np.savez(op.join(dest_folder, 'data.npz'),
-             x_train=x_vals[:split_index, ...],
-             y_train=y_vals[:split_index, ...],
-             x_test=x_vals[split_index:, ...],
-             y_test=y_vals[split_index:, ...])
+
+    if len(split_vals) == 2:
+        np.savez(op.join(dest_folder, 'data.npz'),
+                 x_train=x_vals_split_lst[0],
+                 y_train=y_vals_split_lst[0],
+                 x_test=x_vals_split_lst[1],
+                 y_test=y_vals_split_lst[1])
+
+    if len(split_vals) == 3:
+        np.savez(op.join(dest_folder, 'data.npz'),
+                 x_train=x_vals_split_lst[0],
+                 y_train=y_vals_split_lst[0],
+                 x_test=x_vals_split_lst[1],
+                 y_test=y_vals_split_lst[1],
+                 x_val=x_vals_split_lst[2],
+                 y_val=y_vals_split_lst[2])
```

Comment:
I believe you can create a dict with all the splits and pass it to `np.savez`. This would save some code and handle any arbitrary number of splits/names. This needs testing, but roughly that'd look like:

```
save_dict = {}
for si, split_name in enumerate(split_names):
    save_dict[f'x_{split_name}' = x_vals_split_lst[si]
    save_dict[f'y_{split_name}' = y_vals_split_lst[si]

np.savez(op.join(dest_folder, 'data.npz'), **save_dict)
```